Src/library/base/elks/kernel/type.e

```diff
@@ -1,12 +1,13 @@
 ﻿note
 	description: "Representation of an Eiffel type."
+	remark: "At any given time, there is no more than one instance of TYPE representing a given Eiffel type."
```

Comment:
The only issue with a non-singleton reference type solution is the need to use ~ instead of =. As pointed out in the development mailing list, other solutions have several issues that would require much more work to achieve their goals and might still remain rather fragile. From pragmatic point of view, absence of the singleton requirement is SCOOP-compatible, more storable-friendly, has simpler implementation and is forward-compatible with singleton-based and expanded-type-based solutions if they ever become feasible.